### ISSUE TITLE
genoconvert module

### ISSUE THREAD

**Issue by nevrome** [poseidon-framework/poseidon-hs#12]
Trident can deal equally well with plink binary and eigenstrat data. We could have a module that converts the genotype data in packages from one format to the other. That would also include replacing the format definition in the POSEIDON.yml file.

**Comment by nevrome** [comment]
In this context `forge` should also get an option to define the desired output format

**Comment by stschiff** [comment]
Indeed! However, my Plink module in `sequence-formats` doesn't feature Plink writing yet, just reading, so it'll have to wait until I get around implementing that.

**Comment by stschiff** [comment]
OK, this is partly solved now with forge being able to output PLINK format.

**Comment by nevrome** [comment]
Indeed `forge` covers that usecase then. But I think for ease of use we should add an explicit `convert` module. It would be a confusing user experience to go through `forge` if you just want to "quickly convert this package".

**Comment by stschiff** [comment]
Hmm. What interface do you have in mind? Do you want to convert in place? That sounds dangerous... or do you want to just add the converted genotype files to the same package? That would be possible, but the new genotype datafiles would then not be covered by the `POSEIDON.yml` file (cause that can have only one genotype dataset)...

So I don't really see how this would be any different from a new forge command. With the interface I have in mind it'd be very similar:
```
trident convert -d basedir -n package_name --toEigenstrat -n new_pac_name -o /path/to/new/pac
```
versus
```
trident forge -d basedir -f "*package_name*" --eigenstrat -n new_pac_name -o /path/to/new/pac
```

or is there a cleverer, more intuitive interface?

**Comment by nevrome** [comment]
Hm - I really imagined simple replacement. This conversion from EIGENSTRAT to PLINK and back is lossless, right? And you can only perform it, if you have writing permissions for the respective directories. So while I understand your concern, I also love how simple this would make things:

```
trident convert -d . --outFormat EIGENSTRAT
```

**Comment by stschiff** [comment]
Hmm, OK, that's simple enough. I agree... and with `-d` in this case you would select a single directory? What happens if you run that on the entire repo? Does it then convert all packages?

**Comment by nevrome** [comment]
It does. Probably we should ask the user for confirmation. 

And we should think carefully about the order of events in `convert` to prevent loss of data even in case of an interrupt: 
1. Conversion and therefore creation of new data files
2. Replacement of the paths in the POSEIDON.yml to point to the new data files
3. Finally: Delete the "old" data files

**Comment by stschiff** [comment]
Or perhaps even not deleting the old files... simply pointing the paths in the `POSEIDON.yml` to the new files without deleting the old ones. That would allow for maximum flexibility then to use the data with tools that require either format.

**Comment by nevrome** [comment]
Ah - ja - that's also possible. What should then happen when the conversion is performed in the other direction?

**Comment by nevrome** [comment]
Implemented in #62
